src/model.jl

```diff
@@ -190,6 +190,369 @@ function _make_sets(case::Case, storage::Union{Storage,Nothing})::Sets
 end
 
 
+function _add_constraint_power_balance!(
+    m::JuMP.Model,
+    case::Case,
+    sets::Sets,
+    storage::Storage,
+    demand_flexibility::DemandFlexibility,
+    start_index::Int,
+    interval_length::Int,
+    demand_scaling::Number,
+    load_shed_enabled::Bool,
+)
+    # Generator topology matrix
+    gen_map = _make_gen_map(case)
+    # Branch connectivity matrix
+    branch_map = _make_branch_map(case)
+    # Demand by bus
+    bus_demand = _make_bus_demand(case, start_index, start_index + interval_length - 1)
+    bus_demand *= demand_scaling
+
+    gen_injections = JuMP.@expression(m, gen_map * m[:pg])
+    line_injections = JuMP.@expression(m, branch_map * m[:pf])
+    injections = JuMP.@expression(m, gen_injections + line_injections)
+    if load_shed_enabled
+        injections = JuMP.@expression(m, injections + sets.load_bus_map * m[:load_shed])
+    end
+    withdrawals = JuMP.@expression(m, bus_demand)
+    storage_enabled = (sets.num_storage > 0)
```

Comment:
I thought about that. I guess it's a question of whether we want minimal inputs (i.e. infer whether storage is enabled from `sets`) or minimal code (avoid this line because we already know from the parameter being explicitly passed). I'm agnostic on this, and would follow your and @BainanXia's advice here.

Reply:
I hadn't considered the minimal input vs minimal code tradeoff. Either way is fine to me. What I was thinking when I left the comment was that passing in `storage_enabled` would reduce the number of computations (though only marginally so, I imagine). If we really wanted to reduce the number of times that `storage_enabled` is calculated, we could probably add an `enabled` member to the `Storage` struct (similar to what we do with `DemandFlexibility`).

While this is similar to the next two comments, I believe we would have input savings by passing in a once-calculated `bus_demand`.

Reply:
I think for a simple and compact model, we can implement in a way that it could infer what to do based on the inputs, however, as we keep adding new features to our model, it would be better to let user be explicit on what he/she would like to do via more input parameters. Hence, I lean towards the latter at this point. Any inputs from @jon-hagg @rouille ?

Reply:
Long-term, I think `Storage.enabled` is definitely the way to go.

Reply:
`Storage.enabled` has been implemented.